packages/react-dom/src/__tests__/ReactDOMFizzSuppressHydrationWarning-test.js

```diff
@@ -248,9 +248,10 @@ describe('ReactDOMFizzServerHydrationWarning', () => {
         ]);
       }).toErrorDev(
         [
+          'Expected server HTML to contain a matching <span> in <span>',
           'An error occurred during hydration. The server HTML was replaced with client content in <div>.',
```

Comment:
You still have these though. Isn't it better to have the context here for the ones we check in prod anyway? Is it worth having both? Is it confusing that there are two with subtly different timing?

Maybe it would make more sense to have the DEV only error if it was part of a larger diff across multiple mismatches but right now these are 1:1.

Reply:
In other words, should we just move the insert/deletion errors to the onRecoverableError?

Reply:
I have another PR in the works for the diff. It's not coalesced though.

Reply:
But in either case I think we should fix up the inconsistency first. I.e. your concerns don't seem related to this PR per se, it just aligns the behavior for suppressHydration ones with the rest.